R/vignettes.R

```diff
@@ -146,13 +155,121 @@
     return(invisible())
   }
 
-  file_to_update <- switch(
-    doctype,
-    "docute" = fs::path_abs("docs/index.html", start = path),
-    "docsify" = fs::path_abs("docs/_sidebar.md", start = path),
-    "mkdocs" = fs::path_abs("docs/mkdocs.yaml", start = path)
-  )
-  cli::cli_alert_warning(
-    cli::style_bold("Don't forget to check that vignettes are correctly included in {.file {file_to_update}}.")
-  )
+  if (!nrow(vignettes_titles) >= 1) {
+    return(invisible())
+  }
+
+  if (doctype == "docute") {
+
+    original_index <- .readlines(fs::path_abs("docs/index.html", start = path))
+
+    if (any(grepl("title: \"Articles\"", original_index))) {
+      cli::cli_alert_info("New vignettes were not added automatically in {.file {'docs/index.html'}}. You need to do it manually.")
+      return(invisible())
+    }
+
+    home_line <- which(grepl("\\{title: 'Home', link: '/'\\}", original_index))
+
+    original_index[home_line] <- paste0(
+      original_index[home_line],
+      "\n{
+					   title: \"Articles\",
+					   children:",
+      jsonlite::toJSON(vignettes_titles, pretty = TRUE),
+      "},\n"
+    )
+
+    writeLines(original_index, fs::path_abs("docs/index.html", start = path))
+
+  } else if (doctype == "docsify") {
+
+    original_sidebar <- .readlines(fs::path_abs("docs/_sidebar.md", start = path))
+
+    # Remove the articles / vignettes section to avoid duplicates
```

Comment:
What happens if the user has customized this section? Will it get overwritten automatically? Ideally, we want to populate automatically if there is no user-supplied info, but we don't want things to be overwritten if the user set them up manually.